packages/legend-application-repl/src/components/grid/ServerSideDataSource.ts

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {
+  IServerSideDatasource,
+  IServerSideGetRowsParams,
+} from '@ag-grid-community/core';
+import {
+  TDSGroupby,
+  TDSAggregation,
+  TDSFilter,
+  TDSRequest,
+  TDSSort,
+  TDSFilterCondition,
+  TDS_FILTER_GROUP,
+} from './TDSRequest.js';
+import {
+  guaranteeNonNullable,
+  type GeneratorFn,
+  assertErrorThrown,
+  LogEvent,
+} from '@finos/legend-shared';
+import { LEGEND_REPL_EVENT } from '../../Const.js';
+import {
+  type TDSRowDataType,
+  getTDSRowData,
+  getTDSSortOrder,
+  getTDSFilterOperation,
+  getAggregationFunction,
+  getFilterColumnType,
+} from './GridUtils.js';
+import type { REPLGridClientStore } from '../../stores/REPLGridClientStore.js';
+import { flow, flowResult, makeObservable } from 'mobx';
+import type { INTERNAL__TDSColumn, PRIMITIVE_TYPE } from '@finos/legend-graph';
+
+export class ServerSideDataSource implements IServerSideDatasource {
+  executions = 0;
+  rowData: TDSRowDataType[] = [];
+  columns: INTERNAL__TDSColumn[] = [];
+  editorStore?: REPLGridClientStore | undefined;
+
+  constructor(
+    rowData?: TDSRowDataType[] | undefined,
+    columns?: INTERNAL__TDSColumn[] | undefined,
+    editorStore?: REPLGridClientStore | undefined,
+  ) {
+    makeObservable(this, {
+      fetchRows: flow,
+    });
+    this.rowData = rowData ?? [];
+    this.columns = columns ?? [];
+    this.editorStore = editorStore;
+  }
+
+  *fetchRows(
+    params: IServerSideGetRowsParams<unknown, unknown>,
+  ): GeneratorFn<void> {
+    try {
+      if (this.executions > 0) {
+        if (this.editorStore) {
+          const request = this.extractRequest(params);
```

Comment:
i see no tests in this PR :). probably a good place to start is the extracting of the request from the ag grid params.

Reply:
Will address that in a different commit